File: back/src/main/java/com/haruon/groupware/domain/empInfo/emp/PositionCode.java
package com.haruon.groupware.domain.empInfo.emp;

import lombok.Getter;

@Getter
public enum PositionCode {
    NONE(0),
    INTERN(1),
    STAFF(2),
    SENIOR_STAFF(3),
    ASSISTANT_MANAGER(4),
    MANAGER(5),
    SENIOR_MANAGER(6),
    DIRECTOR(7),
    EXECUTIVE(8);

    private final int level;

    PositionCode(int level) {
        this.level = level;
    }

}

File: front/src/features/employee/registration/model/positionCode.ts
export type PositionCode =
  | 'NONE'
  | 'INTERN'
  | 'STAFF'
  | 'SENIOR_STAFF'
  | 'ASSISTANT_MANAGER'
  | 'MANAGER'
  | 'SENIOR_MANAGER'
  | 'DIRECTOR'
  | 'EXECUTIVE'

export const positionLabels: Record<PositionCode, string> = {
  NONE: '미지정',
  INTERN: '인턴',
  STAFF: '사원',
  SENIOR_STAFF: '주임',
  ASSISTANT_MANAGER: '대리',
  MANAGER: '과장',
  SENIOR_MANAGER: '차장',
  DIRECTOR: '부장',
  EXECUTIVE: '임원',
}
